src/integer/z/fmpz_helpers.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains helpful functions on [`fmpz`].
+
+use super::Z;
+use flint_sys::fmpz::{fmpz, fmpz_abs, fmpz_cmpabs, fmpz_sub};
+
+/// Efficiently finds maximum absolute value and returns
+/// a cloned [`Z`] instance out of a vector of [`fmpz`] instances.
+///
+/// Parameters:
+/// - `fmpz_vector`: contains a number of [`fmpz`] instances
+///
+/// Returns the maximum absolute value out of the given [`fmpz`] vector
+/// as a cloned [`Z`] instance.
+///
+/// # Example
+/// ```compile_fail
+/// use flint_sys::fmpz::fmpz;
+/// use qfall_math::integer::{fmpz_helpers::find_max_abs, Z};
+///
+/// let fmpz_vec = vec![fmpz(0), fmpz(-13), fmpz(10)];
+///
+/// let abs_max = find_max_abs(&fmpz_vec);
+/// assert_eq!(Z::from(13), fmpz_vec);
+/// ```
+pub(crate) fn find_max_abs(fmpz_vector: &Vec<fmpz>) -> Z {
+    // find maximum of absolute fmpz entries
+    let mut max = &fmpz(0);
```

Comment:
The comment seems unnecessary

Reply:
I would like to see the comment since there is an `fmpz` created that is never cleared. This is suspicious behavior and something to watch out for during memory leak hunting. The comment makes it easier to understand and as a result saves time while memory leak hunting.